### PR TITLE
feat: add system setting api

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
- add system setting api
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+@EqualsAndHashCode
+@AllArgsConstructor
+@NoArgsConstructor
+public class DockerImage {
+
+    String registry;
+    String image;
+
+    private static final Pattern PATTERN_IMAGE_FULL = Pattern.compile(
+            "^(([a-z0-9-]+\\.[a-z0-9.]+)(:(\\d{2,5}))?)?\\/?(.+)$");
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java:41]: https://stackoverflow.com/questions/37861791/how-are-docker-image-names-parsed
> TL;DR: The hostname must contain a . dns separator, a : port separator, or the value "localhost" before the first /. Otherwise the code assumes you want the default registry, Docker Hub.

1. Do we support the `localhost` domain?
2. Do we validate the original docker image name? 
    * If not, should we just split the image name with a slash and check if the first part is a valid domain?

Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java]: done

File: server/controller/src/test/java/ai/starwhale/mlops/common/DockerImageTest.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        Assertions.assertEquals("131.0.1.8:5000/star-whale/starwhale:0.3.0-rc.6-nightly-20220920-016d5507",
+                dockerImage.resolve("131.0.1.8:5000/"));
+        Assertions.assertEquals("131.0.1.8:5000/star-whale/starwhale:0.3.0-rc.6-nightly-20220920-016d5507",
+                dockerImage.resolve("131.0.1.8:5000"));
+        Assertions.assertEquals(
+                "homepage-ca.intra.starwhale.ai:5000/star-whale/starwhale:0.3.0-rc.6-nightly-20220920-016d5507",
+                dockerImage.resolve("homepage-ca.intra.starwhale.ai:5000"));
+        Assertions.assertEquals(
+                "homepage-ca.intra.starwhale.ai:5000/star-whale/starwhale:0.3.0-rc.6-nightly-20220920-016d5507",
+                dockerImage.resolve("homepage-ca.intra.starwhale.ai:5000/"));
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/test/java/ai/starwhale/mlops/common/DockerImageTest.java]: done

File: server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    private static final String SLASH = "/";
+
+    public String resolve(String newRegistry) {
+        if (!StringUtils.hasText(newRegistry)) {
+            return image;
+        }
+        if (newRegistry.endsWith(SLASH)) {
+            newRegistry = removeEndingSlash(newRegistry);
+        }
```
[the remark this answers is not in the document]
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/common/DockerImage.java]: done

**Review by jialeicui — APPROVED** [review]
